src/router.h

```diff
@@ -83,19 +83,18 @@ class StreamRouter {
     port_ = port;
     n_threads_ = n_threads;
     timeout_ms_ = timeout_ms;
-  };
-  void start() {
     server_ = std::make_unique<WebsocketServer>(
         [this](WebsocketClient &client, std::vector<uint8_t> data, int type) {
           on_data(client, data, type);
         },
         [this](WebsocketClient &client) { on_connect(client); },
         [this](WebsocketClient &client) { on_disconnect(client); }, port_,
         host_, n_threads_, timeout_ms_);
-
+  };
```

Comment:
This commit breaks the logic of C++, to my taste.

Why have an initialized but not-running websocket at all? It's just introducing extra room for error, no?

Reply:
After discussion update:
I will add all to initialization list in order to respect RAII =)

Reply:
Resolving this one!